packages/mui-material/src/Accordion/Accordion.js

```diff
@@ -115,7 +115,11 @@ const AccordionRoot = styled(Paper, {
 );
 
 const Accordion = React.forwardRef(function Accordion(inProps, ref) {
-  const props = useThemeProps({ props: inProps, name: 'MuiAccordion' });
+  const props = useThemeProps({
+    props: inProps,
+    name: 'MuiAccordion',
+    mergePropsWhiteList: ['TransitionProps'],
```

Comment:
If we are considering this approach of passing through component-level, IMO we should also add a test for Accordion that `TransitionProps` gets merged. This is to prevent bug in case this line is removed in future.

Reply:
I renamed mergePropsWhiteList to shallowMergeProps and also created a test for checking the prop merge.